classes/form/CustomerForm.php

```diff
@@ -138,62 +138,11 @@ public function validate()
             );
             $birthdayField->setValue($dateBuilt->format('Y-m-d'));
         }
-        $this->validateFieldsLengths();
         $this->validateByModules();
 
         return parent::validate();
     }
 
-    protected function validateFieldsLengths()
-    {
-        $this->validateFieldLength('email', 255, $this->getEmailMaxLengthViolationMessage());
-        $this->validateFieldLength('firstname', 255, $this->getFirstNameMaxLengthViolationMessage());
-        $this->validateFieldLength('lastname', 255, $this->getLastNameMaxLengthViolationMessage());
-    }
-
-    /**
-     * @param $fieldName
-     * @param $maximumLength
-     * @param $violationMessage
-     */
-    protected function validateFieldLength($fieldName, $maximumLength, $violationMessage)
-    {
-        $emailField = $this->getField($fieldName);
-        if (strlen($emailField->getValue()) > $maximumLength) {
-            $emailField->addError($violationMessage);
-        }
-    }
-
-    /**
-     * @return mixed
-     */
-    protected function getEmailMaxLengthViolationMessage()
-    {
-        return $this->translator->trans(
-            'The %1$s field is too long (%2$d chars max).',
-            ['email', 255],
-            'Shop.Notifications.Error'
-        );
-    }
-
-    protected function getFirstNameMaxLengthViolationMessage()
-    {
-        return $this->translator->trans(
-            'The %1$s field is too long (%2$d chars max).',
-            ['first name', 255],
-            'Shop.Notifications.Error'
-        );
-    }
-
-    protected function getLastNameMaxLengthViolationMessage()
-    {
-        return $this->translator->trans(
-            'The %1$s field is too long (%2$d chars max).',
-            ['last name', 255],
-            'Shop.Notifications.Error'
-        );
-    }
-
```

Comment:
- These methods are no longer necessary due to
https://github.com/PrestaShop/PrestaShop/pull/19449
- The properties of a field (validation, max length ...) must come from the definition of the class. See https://github.com/PrestaShop/PrestaShop/blob/develop/classes/ObjectModel.php#L153
-   Problems with translating field labels in error messages : 

Example : 
```php
protected function getLastNameMaxLengthViolationMessage()	
    {	
        return $this->translator->trans(	
            'The %1$s field is too long (%2$d chars max).',	
            ['last name', 255],	
            'Shop.Notifications.Error'	
        );	
    }
```

In french it will give : 

``` Le champs last name est trop long. (Longeur maximale : 255)```